src/Layer.js

```diff
@@ -305,8 +350,49 @@ export default class Layer {
         return this._integrator;
     }
 
+    /**
+     *  Get the features in a position
+     * @param  {object} pos
+     * @param  {number} pos.x
+     * @param  {number} pos.y
+     * @memberof carto.Layer
+     * @instance
+     * @api
+     */
     getFeaturesAtPosition (pos) {
-        return this._renderLayer.getFeaturesAtPosition(pos).map(this._addLayerIdToFeature.bind(this));
+        return this.visible
+            ? this._renderLayer.getFeaturesAtPosition(pos).map(this._addLayerIdToFeature.bind(this))
+            : [];
+    }
+
+    /**
+     * Change layer visibility to visible
+     *
+     * @memberof carto.Layer
+     * @instance
+     * @api
+     *
+     * @fires updated
+     */
+    show () {
+        this.visible = true;
+        this._integrator.changeVisibility(this._id, this.visibility);
+        this._fire('updated', { visibility: this.visibility });
```

Comment:
Do we really need to put the visibility in the event? this change requires to change the documentation and to fire with the same property in all points of the program. I would prefer to keep it simple stupid (KISS)